Xamarin.Forms.Platform.iOS/Renderers/ListViewRenderer.cs

```diff
@@ -91,24 +92,25 @@ public override void LayoutSubviews()
 
 		protected override void Dispose(bool disposing)
 		{
-			// check inset tracker for null to 
-			if (disposing && _insetTracker != null)
+			if (_disposed)
+				return;
+
+			if (disposing)
 			{
-				_insetTracker.Dispose();
-				_insetTracker = null;
+				if (_insetTracker != null)
+				{
+					_insetTracker.Dispose();
+					_insetTracker = null;
+				}
 
 				var viewsToLookAt = new Stack<UIView>(Subviews);
 				while (viewsToLookAt.Count > 0)
 				{
-					var view = viewsToLookAt.Pop();
-					var viewCellRenderer = view as ViewCellRenderer.ViewTableCell;
-					if (viewCellRenderer != null)
-						viewCellRenderer.Dispose();
-					else
-					{
-						foreach (var child in view.Subviews)
-							viewsToLookAt.Push(child);
-					}
+					UIView view = viewsToLookAt.Pop();
+					foreach (UIView child in view.Subviews)
+						viewsToLookAt.Push(child);
+
+					view.Dispose();
```

Comment:
As this loop executes, eventually we hit `CellTableViewCell`. Calling `Dispose` on it should properly release the reference to its `_cell`.

The original code was calling `Dispose` only on `ViewCellRenderer.ViewTableCell` instances. At the moment, I'm not sure if disposing every subview is the correct approach, so this is up for discussion. Please note that `TableViewRenderer` is doing the same thing, so it might need an update as well.